crypto/keys/src/ed25519.rs

```diff
@@ -192,11 +190,11 @@ impl Verifier<Ed25519Signature> for Ed25519Public {
 }
 
 impl TryFrom<&[u8]> for Ed25519Public {
-    type Error = SignatureError;
+    type Error = KeyError;
```

Comment:
This change seems insubstantial, and it makes it much much easier to do generic code with `Kex` when you know the error type here will be `KeyError`